### PR TITLE
A simple working modal

### PR DESCRIPTION
**Pull request by cybadave**
The Overlay component's dependence on react-tippy doesn't play nicely with my modal use-cases. Here's a simple modal for anyone else who needs one.

### CODE REVIEW

File: components/overlays/Modal.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React from "react"
+import Portal from "react-portal"
+
+export default class Modal extends React.Component {
+    static defaultProps = {
+        handleOutsideClick: () => {},
+        renderFunc: () => {},
+        open: false,
+    }
```
Comment (diogobeda) [inline components/overlays/Modal.js]: Some PropTypes for documentation? :)

File: components/overlays/Modal.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import React from "react"
+import Portal from "react-portal"
```
Comment (diogobeda) [inline components/overlays/Modal.js]: Very much 💅  here, but missing `;` and single-quotes. Consider installing `prettier` on your editor so you don't have to worry about this stuff ;)

File: components/overlays/Modal.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            if (this.wrapper && !this.wrapper.node.contains(event.target)) {
+                this.props.handleOutsideClick();
+            }
+        }
+    }
+
+    render() {
+        return (
+            <Portal ref={node => { this.wrapper = node; }} isOpened={this.props.open}>
+                {this.props.renderFunc()}
```
Comment (diogobeda) [inline components/overlays/Modal.js]: What do you think about using `children` or plain simple `render`? I see that more often on other components like: formik, downshift. Check out this article:
https://blog.kentcdodds.com/how-to-give-rendering-control-to-users-with-prop-getters-549eaef76acf

File: components/overlays/Modal.js (hunk 2 above)
Comment (cybadave) [reply components/overlays/Modal.js]: :+1:  Prettier was just not set up properly with the m-dash-ui project - setup now.

File: components/overlays/Modal.js (hunk 1 above)
Comment (cybadave) [reply components/overlays/Modal.js]: :+1:

File: components/overlays/Modal.js (hunk 3 above)
Comment (cybadave) [reply components/overlays/Modal.js]: This doesn't work with my use-case, but it's a great idea - I've made it possible to pass either nodes or a function to render the children.

**Pull request merged**